src/lib/operators/aggregate.cpp

```diff
@@ -605,14 +605,19 @@ write_aggregate_values(std::shared_ptr<ValueColumn<AggregateType>> column,
   auto& values = column->values();
   auto& null_values = column->null_values();
 
+  values.resize(results->size());
```

Comment:
I am completely not sure about this, but wouldn't it be better to use `reserve()` instead of `resize()` here, and further down keep the `push_back()`? `resize()` creates a bunch of default constructed `AggregateType()`s here, which we later replace by the actual values. `reserve()` would just allocate the memory, and we then insert the actual objects only ones.

Same for all the other `resize()`s further down.

Reply:
`reserve` <3

Reply:
Would there be measurable performance improvement, @lanice ? (your point sounds reasonable, just asking out of curiosity)

Reply:
Yes, but the other way round:
http://quick-bench.com/gpuVZCbF79ZeHtbtUNvInt-qqVY

With a non-trivial constructor, this slightly changes, but resize is still faster.

Reply:
wow, the more you know.

Reply:
Fu, forgot about the part where emplace/push_back need to check whether sufficient memory is available on every call....

Reply:
...and move the end pointer

Reply:
Shouldn't `reserve()` and `[]` be safe? That's apparently the fastest option: http://quick-bench.com/mHO04EAn5UdyQZUBd6Jgqqfh-XA

But I have no idea what I am doing here. :)

Reply:
reserve() and [] doesn't move the end-ptr, so the vector would, for example, report the wrong size

Reply:
would be cool if there was some way of doing resize without the invocations of the default constructor.

Reply:
Interesting!
So `reserve()` and `[]` would be fastest, but without an additional call to `resize()` afterwards leaves us with a somewhat broken vector.

I don't understand why `resize()` after `reserve()` + `[]` is so much slower than just `reserve()` + `[]`:
http://quick-bench.com/bW_6Msgt0oImmWr4JWp2_1u68BA

Shouldn't it do basically nothing, and just move the end pointer and adjust the size?

Anyway, `resize()` + `[]` is our best option for now then?

Reply:
As far as the vector knows, the memory that has been reserved is still uninitialized.

Greenlight anyone?